### ISSUE TITLE
What is the minimal C# Language version of the generated source code?

### ISSUE THREAD

**Issue by kosalanuwan** [martinothamar/Mediator#95]
I am trying to replace MediatR with this source generating Mediator, but it seems the generated source code is throwing some compilation errors. One thing that I noticed is that the dev environment I am in has only and scoped to .NET Core 3.1 SDK and that limits me to a certain C# language features and wonder what is the minimal C# language version that is required to compile the generated source code?

**Comment by martinothamar** [comment]
Hi! The library targets `netstandard2.1` currently, which NET Core 3.1 supports. NET Core 3.1 is however out of support by Microsoft (as of December 13, 2022), so it's not really recommended to use as it won't get security patches etc..

In addition, the source generator needs a relatively new version of the Roslyn libraries that come with the .NET SDK. We have support for Roslyn 3.8+, but .NET Core 3.1 _SDK_ seems to bring something older than that. So the SDK itself won't work. But if you have the .NET 6.0 SDK and use `netcoreapp3.1` target framework you will actually be able to build. Atleast I managed to using [this patch (b1a560d)](https://github.com/martinothamar/Mediator/commit/b1a560d06cf5021276891fc4612798e46b2d41af) which I'll publish to the 2.1 now. The Mediator library also references the Microsoft DependencyInjection libraries 6.0+, so if you go with `netcoreapp3.1` you still will have to use a newer version of the DI libraries. 

**TLDR**: you can use `netcoreapp3.1` target framework if you build with >= 6.0 NET SDK, as long as you reference MS DI libraries >= 6.0, but since .NET Core 3.1 is out of support, I highly recommend you switch over to .NET 6+
